### PR TITLE
ddl: truncate table/partition should clear the table tiflash replica available status

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs <crazycs520@gmail.com>

### What problem does this PR solve? 

When the table has been truncated, we should clear the TiFlash replica available status and wait TiFlash replica resynchronization.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

Release note

 - Fix the issue that truncate table/partition doesn't clear the table TiFlash replica available status

### CODE REVIEW

File: ddl/db_test.go [hunk 1]
```diff
@@ -2598,6 +2598,49 @@ func (s *testDBSuite3) TestTruncateTable(c *C) {
...
+	err = domain.GetDomain(tk.Se).DDL().UpdateTableReplicaInfo(tk.Se, partition.Definitions[0].ID, true)
+	c.Assert(err, IsNil)
+	err = domain.GetDomain(tk.Se).DDL().UpdateTableReplicaInfo(tk.Se, partition.Definitions[1].ID, true)
+	c.Assert(err, IsNil)
+	t1 = testGetTableByName(c, s.s, "test", "t1")
+	c.Assert(t1.Meta().TiFlashReplica, NotNil)
+	c.Assert(t1.Meta().TiFlashReplica.Available, IsTrue)
+	c.Assert(t1.Meta().TiFlashReplica.AvailablePartitionIDs, DeepEquals, []int64{partition.Definitions[0].ID, partition.Definitions[1].ID})
+
+	tk.MustExec("alter table t1 truncate partition p0")
```
Comment (djshow832) [inline ddl/db_test.go]: Try to truncate it twice.

Comment (crazycs520) [reply ddl/db_test.go]: done.

**Pull request merged**